src/vixen/core.clj

```diff
@@ -144,13 +144,14 @@
            [[n p] & others] [[node path]]]
       (if n
         (let [newly (selector n p vix)]
-          (recur (if (empty? newly) found (concat found newly))
+          ;; Prefer `into` over `concat` below to support arbitrary stack depth and result count
+          (recur (if (empty? newly) found (into found newly))
                  (if (empty? (:content n))
                              others
-                             (concat others
-                                     (map-indexed (fn [i child]
-                                                    [child (conj p i)])
-                                                  (:content n))))))
+                             (into others
```

Comment:
I *think* you need to force others to a vector here or this collection is in the reverse order it was prior to the change. `(into (vec others) ...)`